uarray/_backend.py

```diff
@@ -537,3 +539,137 @@ def __ua_convert__(self, dispatchables, coerce):
         return converted
 
     return __ua_convert__
+
+
+def determine_backend(value, dispatch_type, *, domain, only=True, coerce=False):
+    """Set the backend to the first active backend that supports ``value``
+
+    This is useful for functions that call multimethods without any
+    dispatchable arguments. You can use `determine_backend` to ensure the same
+    backend is used within a block of multimethod calls.
+
+    Parameters
+    ----------
+    value
+        The value being tested
+    dispatch_type
+        The dispatch type associated with ``value``, aka
+        ":ref:`marking <MarkingGlossary>`".
+    domain: string
+        The domain to query for backends and set.
+    coerce: bool
+        Whether or not to allow coercion to the backend's types. Implies ``only``.
+    only: bool
+        Whether or not this should be the last backend to try.
+
+    See Also
+    --------
+    set_backend: For when you know which backend to set
+
+    Notes
+    -----
+
+    Support is determined by the ``__ua_convert__`` protocol. Backends not
+    supporting the type must return ``NotImplemented`` from their
+    ``__ua_convert__`` if they don't support input of that type.
+
+    Examples
+    --------
+
+    Suppose we have two backends ``BackendA`` and ``BackendB`` each supporting
+    different types, ``TypeA`` and ``TypeB``. Neither supporting the other type:
+
+    >>> with ua.set_backend(BackendA):
+    ...     call_multimethod(TypeB(), TypeB())
+    Traceback (most recent call last):
+        ...
+    uarray.BackendNotImplementedError: ...
+
+    Now consider a multimethod that creates a new object of ``TypeA``, or
+    ``TypeB`` depending on the active backend.
+
+    >>> with ua.set_backend(BackendA), ua.set_backend(BackendB):
+    ...         res = creation_multimethod()
+    ...         call_multimethod(res, TypeA())
+    Traceback (most recent call last):
+        ...
+    uarray.BackendNotImplementedError: ...
+
+    ``res`` is an object of ``TypeB`` because ``BackendB`` is set in the
+    innermost with statement. So, ``call_multimethod`` fails since the types
+    don't match.
+
+    Instead, we need to first find a backend suitable for all of our objects.
+
+    >>> with ua.set_backend(BackendA), ua.set_backend(BackendB):
+    ...     x = TypeA()
+    ...     with ua.determine_backend(x, "mark", domain="ua_examples"):
+    ...         res = creation_multimethod()
+    ...         type(call_multimethod(res, x))
+    TypeA
+
+    """
+    dispatchables = (Dispatchable(value, dispatch_type, coerce),)
+    backend = _uarray.determine_backend(domain, dispatchables, coerce)
+
+    return set_backend(backend, coerce=coerce, only=only)
+
+
+def determine_backend_multi(*dispatchables, domain, only=True, coerce=False):
```

Comment:
Can we add `mark=` here in case one didn't pass in `ua.Dispatchable`s? And can we change the input to an explicit iterable instead of `*dispatchables`?

Reply:
Did the second, not so sure about the first. A single `mark=` argument is insufficient to cover all cases. e.g. if you have dtype dispatching you need two different dispatch types.

Reply:
So, if the `mark=` is passed, it'd automatically mark all non-dispatchables.

Reply:
It'd still be nice to see this addressed, see my edit to my previous comment on how to make it general.